server/internal/core/application/covenantless.go

```diff
@@ -1080,16 +1078,24 @@ func (s *covenantlessService) startFinalization() {
 	s.forfeitTxs.init(connectors, requests)
 
 	if len(vtxoTree) > 0 {
-		log.Debugf("signing vtxo tree for round %s", round.Id)
 
-		signingSession := newMusigSigningSession(len(cosigners))
+		uniqueSignerPubkeys := make(map[string]struct{})
+		for _, request := range requests {
+			for _, pubkey := range request.SignerPubKeys {
+				uniqueSignerPubkeys[pubkey] = struct{}{}
+			}
+		}
+
+		nbOfCosigners := len(uniqueSignerPubkeys)
```

Comment:
this number should be wrong as all requests contain the server's key, shouldn't it?